NPE Attempt to get length of null array storage.LocalStorage

Taken from Google Play for version 2017.05.09:

```
io.reactivex.exceptions.OnErrorNotImplementedException: Attempt to get length of null array
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:74)
	at io.reactivex.internal.operators.observable.ObservableTakeWhile$TakeWhileObserver.onError(ObservableTakeWhile.java:100)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:276)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:252)
	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:109)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:168)
	at android.app.ActivityThread.main(ActivityThread.java:5845)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:797)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:687)
	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:102)
Caused by: java.lang.NullPointerException: Attempt to get length of null array
	at cgeo.geocaching.storage.LocalStorage$2.call(LocalStorage.java:338)
	at cgeo.geocaching.storage.LocalStorage$2.call(LocalStorage.java:331)
	at io.reactivex.internal.operators.observable.ObservableDefer.subscribeActual(ObservableDefer.java:32)
	at io.reactivex.Observable.subscribe(Observable.java:10841)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:451)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:61)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:52)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:154)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:269)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
```

```
io.reactivex.exceptions.OnErrorNotImplementedException: Attempt to get length of null array
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:704)
	at io.reactivex.internal.functions.Functions$OnErrorMissingConsumer.accept(Functions.java:701)
	at io.reactivex.internal.observers.LambdaObserver.onError(LambdaObserver.java:74)
	at io.reactivex.internal.operators.observable.ObservableTakeWhile$TakeWhileObserver.onError(ObservableTakeWhile.java:100)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.checkTerminated(ObservableObserveOn.java:276)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.drainNormal(ObservableObserveOn.java:172)
	at io.reactivex.internal.operators.observable.ObservableObserveOn$ObserveOnObserver.run(ObservableObserveOn.java:252)
	at io.reactivex.android.schedulers.HandlerScheduler$ScheduledRunnable.run(HandlerScheduler.java:109)
	at android.os.Handler.handleCallback(Handler.java:739)
	at android.os.Handler.dispatchMessage(Handler.java:95)
	at android.os.Looper.loop(Looper.java:158)
	at android.app.ActivityThread.main(ActivityThread.java:7229)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1230)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1120)
Caused by: java.lang.NullPointerException: Attempt to get length of null array
	at cgeo.geocaching.storage.LocalStorage$2.call(LocalStorage.java:338)
	at cgeo.geocaching.storage.LocalStorage$2.call(LocalStorage.java:331)
	at io.reactivex.internal.operators.observable.ObservableDefer.subscribeActual(ObservableDefer.java:32)
	at io.reactivex.Observable.subscribe(Observable.java:10841)
	at io.reactivex.internal.operators.observable.ObservableSubscribeOn$SubscribeTask.run(ObservableSubscribeOn.java:96)
	at io.reactivex.Scheduler$DisposeTask.run(Scheduler.java:451)
	at io.reactivex.internal.schedulers.ScheduledRunnable.run(ScheduledRunnable.java:61)
	at io.reactivex.internal.schedulers.ScheduledRunnable.call(ScheduledRunnable.java:52)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:154)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:269)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
```

2 of these reports, both Android 6.0:
- HTC One M8
- Samsung S7 

No user message.

--- a/main/src/cgeo/geocaching/storage/LocalStorage.java
+++ b/main/src/cgeo/geocaching/storage/LocalStorage.java
@@ -27,6 +27,7 @@ import java.util.regex.Pattern;
 import io.reactivex.Observable;
 import io.reactivex.functions.Consumer;
 import io.reactivex.schedulers.Schedulers;
+import org.apache.commons.lang3.ArrayUtils;
 import org.apache.commons.lang3.CharEncoding;
 import org.apache.commons.lang3.StringUtils;
 
@@ -335,12 +336,14 @@ public final class LocalStorage {
                 final File currentDataDir = new File(getExternalPrivateCgeoDirectory(), GEOCACHE_DATA_DIR_NAME);
                 Log.i("Moving geocache data to " + newDataDir.getAbsolutePath());
                 final File[] files = currentDataDir.listFiles();
-                progress.setMaxProgressAndReset(files.length);
-                progress.setProgress(0);
                 boolean success = true;
-                for (final File geocacheDataDir : files) {
-                    success &= FileUtils.moveTo(geocacheDataDir, newDataDir);
-                    progress.incrementProgressBy(1);
+                if (ArrayUtils.isNotEmpty(files)) {
+                    progress.setMaxProgressAndReset(files.length);
+                    progress.setProgress(0);
+                    for (final File geocacheDataDir : files) {
+                        success &= FileUtils.moveTo(geocacheDataDir, newDataDir);
+                        progress.incrementProgressBy(1);
+                    }
                 }
 
                 Settings.setExternalPrivateCgeoDirectory(newExtDir);

--- a/main/src/cgeo/geocaching/utils/FileUtils.java
+++ b/main/src/cgeo/geocaching/utils/FileUtils.java
@@ -458,8 +458,10 @@ public final class FileUtils {
         if (file.isDirectory()) {
             long result = 0;
             final File[] fileList = file.listFiles();
-            for (final File aFileList : fileList) {
-                result += getSize(aFileList);
+            if (ArrayUtils.isNotEmpty(fileList)) {
+                for (final File aFileList : fileList) {
+                    result += getSize(aFileList);
+                }
             }
             return result; // return the file size
         }

